device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -354,8 +369,13 @@ public void sendReportedProperties(Set<Property> reportedProperties, int version
      */
     public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback callback, Object callbackContext) throws IllegalArgumentException
     {
-        //Codes_SRS_INTERNALCLIENT_34_069: [This function shall register the provided callback and context with its device IO instance.]
-        this.deviceIO.registerConnectionStatusChangeCallback(callback, callbackContext);
+        this.connectionStatusChangeCallback = callback;
+        this.connectionStatusChangeCallbackContext = callbackContext;
+
+        if (this.deviceIO != null)
```

Comment:
can `this.deviceIO` be null? then shouldn't this method throw?

Reply:
I see that it can be null if it's using multiplexing ... ignore my previous commnet.

Reply:
if we had a boolean flag that we would set if multiplexing was registered, we could just check that  ... just a thought

Reply:
We sort of have that flag already at the DeviceClient level in the Enum ```ioTHubConnectionType```

Unfortunately, the internal client level also encompasses the ModuleClient, which has no concept of multiplexing. I'm a bit on the fence about adding a boolean about multiplexing at this level because of that

Reply:
fair enough. just wanted to bring it up if it might add value to not rely on contextual knowledge about how the multiplexing is designed to read the code.

Reply:
How about a method `isClientMultiplexing()` and have the body of that return `this.deviceIO != null`. It reads better, and if you later device to change the condition, you won't have to update it in several places.

Reply:
I've added a ```isMultiplexed``` flag within the InternalClient now to make this stuff more obvious, and so that it can be referenced when we want to block certain things like setting the proxySettings of a multiplexed device